mohanadarafe/GuideMe
# Search bar is now connected to BottomMenu and MoreDetails

## Please Test
Because I made some refactoring in important components (MapData.js), please make sure to test the app and try to break the Search and/or MoreDetails components, I believe in you all! 
## Bug fixed
- Issue #210 as well as #208 
## Implementation
Implemented what was asked in issue #161 
### More
- You can search for a building name, full name, address, services, and departments. When selecting any of them, the MoreDetails will display the building information. 
- Adjusted arrowBackButton in DoubleSearch to stay consistent with PreviewDirections.js and Directions.js.
- Added fullName field in MoreDetails.js
- Added 2 FIXME comments regarding some components that trigger bad warnings. 
## Even More
- I upgraded React Native to the most recent version which is [37.0.0](https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6). Expo is now smoother and faster with a slightly new interface. I have some tricks to test your app faster and with fewer tunnel bugs now (pm me if needed)
- I upgraded Expo CLI to the most recent version.
- The two upgrades were successful in Travis and don't forget to do `npm install`.
- I added devtools dependencies that communicate with the Debugger and set up React Native Debugger. So if someone wants to use the debugger, let me know I could help you since it's local (I can't push a debugger remotely).
## Refactoring
- I refactored MapData.js almost completely.
- I refactored MoreDetails a bit.
- I updated Search and DoubleSearch.
## Test
1. Check out to `feature/#166`
2. Make sure to run `npm install` before running the app with _expo start_
3. Search for anything in the Search Bar.
4. Click on the arrow up from bottomMenu.
## We move 🤝
